### PR TITLE
ovirt-provider-ovn blog: PKI setup

### PR DESCRIPTION
**Pull request by mmirecki**
Signed-off-by: Marcin Mirecki<mmirecki@redhat.com>

Fixes issue # (delete if not relevant)

Changes proposed in this pull request:

-

-

-

I confirm that this pull request was submitted according to the [contribution guidelines](/CONTRIBUTING.md): (please @mention yourself to sign)

This pull request needs review by: (please @mention the reviewer if relevant)

### CODE REVIEW

**Comment by dankenigsberg** [comment]
The commit message needs some content

File: source/blog/2016-11-02-ovirt-provider-ovn.md [hunk 1]
```diff
@@ -204,7 +205,44 @@ If used to set up ovirt-provider-ovn, engine-setup will perform the following ta
 * install ovs/ovn packages
 * install ovirt-provider-ovn package
 * add and configure a default External Network OVN provider. The engine provider will be configured to connect to provider on localhost. The provider will also be configured with the user and password specified during the setup process.
+* configure the OVN and ovirt-provider-ovn public key infrastructure.
```
Comment (dankenigsberg) [inline source/blog/2016-11-02-ovirt-provider-ovn.md]: "the OVN" is a vague term. what are you referring to? ovn-central?

File: source/blog/2016-11-02-ovirt-provider-ovn.md [hunk 2]
```diff
@@ -204,7 +205,44 @@ If used to set up ovirt-provider-ovn, engine-setup will perform the following ta
...
 
+### Configuring PKI manually
+
+The PKI for the provider and driver are configured automatically by engine-setup and vdsm-tool.
```
Comment (dankenigsberg) [inline source/blog/2016-11-02-ovirt-provider-ovn.md]: in first usage, please mention PKI = Public key infrastructure

File: source/blog/2016-11-02-ovirt-provider-ovn.md [hunk 3]
```diff
@@ -51,6 +51,7 @@ After installing the driver and OVN, the OVN-controller must be configured. This
     vdsm-tool ovn-config <OVN central server IP> <local IP used for OVN tunneling>
 
 The second parameter (local IP used for OVN tunneling) can be the IP address of the ovirtmgmt interface on the host, and should be reachable by OVN hosts and central server.
+vdsm-tool will also configure ovn-controller with the PKI required to safely connect to the ovn central server.
```
Comment (mkalfon) [inline source/blog/2016-11-02-ovirt-provider-ovn.md]: Maybe in a new paragraph:
"To add support for Transport Layer Security (TLS) connectivity with the OVN-controller, by default, vdsm-tool creates private and public keys, and configures OVN to enable TLS encryption between the host and the central OVN-controller."

Marcin, do you have a patch which includes the changes in the script?

File: source/blog/2016-11-02-ovirt-provider-ovn.md (hunk 1 above)
Comment (mkalfon) [inline source/blog/2016-11-02-ovirt-provider-ovn.md]: Can you please elaborate what is exactly planned during the installation?

File: source/blog/2016-11-02-ovirt-provider-ovn.md [hunk 4]
```diff
@@ -204,7 +205,44 @@ If used to set up ovirt-provider-ovn, engine-setup will perform the following ta
...
+    ssl_enabled=true # enables/disables https for the REST API
+    key-file=<private key file>
+    cert-file=<certificate file>
+    cacert-file=<ca certificate file>
+
+The PKI of the provider will be used to connect to the OVN north db, as well as for the https connection.
+The CA certificate must used for signing the certificate must be added to the external provider trust store on the engine:
+
+
+    keytool -import -alias <certificate alias> -keystore /var/lib/ovirt-engine/external_keystore -file <ca certificate file> -noprompt -storepass <store password, default: 'changeit'>
```
Comment (mkalfon) [inline source/blog/2016-11-02-ovirt-provider-ovn.md]: default?

File: source/blog/2016-11-02-ovirt-provider-ovn.md (hunk 1 above)
Comment (mmirecki) [inline source/blog/2016-11-02-ovirt-provider-ovn.md]: Added a more detailed description

File: source/blog/2016-11-02-ovirt-provider-ovn.md (hunk 2 above)
Comment (mmirecki) [inline source/blog/2016-11-02-ovirt-provider-ovn.md]: Done

File: source/blog/2016-11-02-ovirt-provider-ovn.md (hunk 3 above)
Comment (mmirecki) [inline source/blog/2016-11-02-ovirt-provider-ovn.md]: Added with some modifications:
- we use the vdsm key/cert for SSL.
- ovn-controller only connects to the south db
- ovn-controller is the part installed on the hosts, there is not central ovn-controller (misleading, I know)

https://gerrit.ovirt.org/#/c/74895

File: source/blog/2016-11-02-ovirt-provider-ovn.md (hunk 4 above)
Comment (mmirecki) [inline source/blog/2016-11-02-ovirt-provider-ovn.md]: This is the default value. Removing
Replacing with a pointer to a config file which contains both the keystore path and password

File: source/blog/2016-11-02-ovirt-provider-ovn.md [hunk 5]
```diff
@@ -51,6 +51,8 @@ After installing the driver and OVN, the OVN-controller must be configured. This
     vdsm-tool ovn-config <OVN central server IP> <local IP used for OVN tunneling>
 
 The second parameter (local IP used for OVN tunneling) can be the IP address of the ovirtmgmt interface on the host, and should be reachable by OVN hosts and central server.
+vdsm-tool will also configure ovn-controller with the PKI required to safely connect to the ovn central server.
+To add support for Transport Layer Security (TLS) connectivity with the OVN-controller, vdsm-tool will use the VDSM private and public keys, and configures OVN-controller to enable PKI encryption between the host and the OVN south db on the OVN central server. This provides privacy and data-integrity in the management connections to the south database
```
Comment (mkalfon) [inline source/blog/2016-11-02-ovirt-provider-ovn.md]: Please add "the" after "configures".

Comment (mmirecki) [inline source/blog/2016-11-02-ovirt-provider-ovn.md]: I'm not really sure about this one.
Is OVN-controller a thing, or is this a name of a thing?
Note that for we do not use "the" for vdsm and ovirt-engine, as these are names. I think the same rule applies here, so I would rather not have it here either.

Comment (dankenigsberg) [inline source/blog/2016-11-02-ovirt-provider-ovn.md]: I'd rather keep the text as it is, without "the".

File: source/blog/2016-11-02-ovirt-provider-ovn.md [hunk 6]
```diff
@@ -204,7 +205,44 @@ If used to set up ovirt-provider-ovn, engine-setup will perform the following ta
 * install ovs/ovn packages
```
Comment (mkalfon) [inline source/blog/2016-11-02-ovirt-provider-ovn.md]: maybe we should mention that engine-setup handles only the installation of OVN packages on the central server.

**Pull request merged**